Automattic/wp-calypso
# Domains: Enable the new "transfer or connect" domain flow

#### Changes proposed in this Pull Request

This PR updates the links that direct users to the "transfer or connect" flow, which is used when they already have a domain they'd like to use with their WPCOM site. These links will now take users to the new updated flow introduced by #55521. E2E domain mapping tests were also updated. This is part of the domain pages redesign work described at pcYYhz-gf-p2.

There are three links that were updated:
- "Already own a domain?" link at the bottom of the domain search page
![Annotation on 2021-08-25 at 16-11-52](https://user-images.githubusercontent.com/5324818/130851219-ff253300-8651-43e4-98c5-5837db97a123.png)
- "Use a domain I own" button at site domains management page when the user already has domains in their site
![Annotation on 2021-08-25 at 16-14-19](https://user-images.githubusercontent.com/5324818/130851252-9f69907b-552e-4ee7-9941-eb6b5c792cd6.png)
- Cards in the site domains management page when the user does not have domains in their site
![Annotation on 2021-08-25 at 16-10-39](https://user-images.githubusercontent.com/5324818/130851275-500efbfe-a359-4990-a592-2485b6b005a3.png)

#### Testing instructions

Open the live Calypso link and check if those three links take you to the new "transfer or connect" flow, which starts in this page:
<img width="1067" alt="Screen Shot 2021-08-25 at 16 16 33" src="https://user-images.githubusercontent.com/5324818/130851447-fe083e17-d7db-45cd-a838-32089c449551.png">
